beacon-chain/blockchain/head.go

```diff
@@ -63,58 +59,6 @@ type head struct {
 	state state.BeaconState            // current head state.
 }
 
-// Determined the head from the fork choice service and saves its new data
-// (head root, head block, and head state) to the local service cache.
-func (s *Service) updateHead(ctx context.Context, balances []uint64) ([32]byte, error) {
```

Comment:
This function is deprecated, it only simply calls to `forkchoice.Head`, as forkchoice now tracks the justified checkpoints and it is guaranteed to have them already (so the call to fill in missing blocks is irrelevant)